### PR TITLE
Add support for decimal round, abs and negate functions

### PR DESCRIPTION
**Pull request by karteekmurthys**
This PR implements Abs and Negate functions as VectorFunctions for decimal type. The existing wrapper for abs() and negate() in ArithmeticImpl.h cannot be extended for int128, hence these functions are customized.

### CODE REVIEW

File: velox/functions/prestosql/DecimalArithmetic.cpp [hunk 1]
```diff
@@ -97,6 +98,52 @@ class DecimalBaseFunction : public exec::VectorFunction {
...
+  DecimalUnaryBaseFunction(uint8_t aRescale) : aRescale_(aRescale) {}
+
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& resultType,
+      exec::EvalCtx& context,
+      VectorPtr& result) const override {
+    auto rawResults = prepareResults(rows, resultType, context, result);
+    if (args[0]->isConstantEncoding()) {
```
Comment (Yuhta) [inline velox/functions/prestosql/DecimalArithmetic.cpp:116]: You need to decode the arguments (see `DecodedArgs`).

File: velox/functions/prestosql/DecimalArithmetic.cpp [hunk 2]
```diff
@@ -97,6 +98,52 @@ class DecimalBaseFunction : public exec::VectorFunction {
   const uint8_t bRescale_;
 };
 
+template <
+    typename R /* Result Type */,
+    typename A /* Argument */,
+    typename Operation /* Arithmetic operation */>
+class DecimalUnaryBaseFunction : public exec::VectorFunction {
```
Comment (Yuhta) [inline velox/functions/prestosql/DecimalArithmetic.cpp:105]: You probably also want to overwrite `supportsFlatNoNullsFastPath()`

File: velox/functions/prestosql/DecimalArithmetic.cpp (hunk 1 above)
Comment (karteekmurthys) [reply velox/functions/prestosql/DecimalArithmetic.cpp:118]: For unary operations, the arguments are not encoded. Refer: https://github.com/facebookincubator/velox/pull/4557#discussion_r1167016474

File: velox/functions/prestosql/DecimalArithmetic.cpp (hunk 2 above)
Comment (karteekmurthys) [reply velox/functions/prestosql/DecimalArithmetic.cpp:105]: Done.

**Comment by karteekmurthys** [comment]
@Yuhta just following up on this.

**Pull request closed without merging**